### ISSUE TITLE
handling of discrepancy between predictorMatrix and method in v3.0

### ISSUE THREAD

**Issue by wepelham3** [stefvanbuuren/mice#93]
Had been getting this error on `mice()` calls in lots of existing code:

```
Error in reformulate(setdiff(vars, j), response = j) : 
  'termlabels' must be a character vector of length at least one
```

I finally determined that it arises when you pass a non-empty imputation method for a column with no missing values.

```
# how many missings in each variable?

lapply(nhanes, function(.x){sum(is.na(.x))})

# runs

mice(nhanes2,
     method = c('sample', 'pmm', 'logreg', 'norm'))

# throws error

mice(nhanes2,
     predictorMatrix = quickpred(nhanes2),
     method = c('sample', 'pmm', 'logreg', 'norm'))

# runs if I set method vector to empty string for age, which has no missing values

mice(nhanes2,
     predictorMatrix = quickpred(nhanes2),
     method = c('' ,'pmm' ,'logreg', 'norm'))
```

If I am understanding issue correctly:

Passing a non-empty imputation method for a variable that had no missing values was OK in previous versions of the package, so perhaps worth noting in the documentation for `method` arg that it now MUST agree.

Or perhaps a check for this prior to imputation that provides informative error message?

**Comment by gerkovink** [comment]
Thanks. Bug solved in Pull Request #94 

All the best, 

Gerko

**Comment by stefvanbuuren** [comment]
Thanks. Now fixed in `mice 3.0.7`.
